hangman.py

```diff
@@ -0,0 +1,338 @@
+    '''
+    Main information for the player
+    '''
+    print("You have", guesses_left, "guesses left")
+    print("Available letters:", get_available_letters(letters_guessed))
+    guessed_letter = input("Please guess a letter: ")
+    return guessed_letter
+
+
+def not_correct_input_losing_warnings(warnings_left, secret_word, letters_guessed):
+    '''
+    System of losing warnings when letter wasn't input
+    '''
+    warnings_left -= 1
+    print("Oops! That is not a valid letter. You have", warnings_left, "warnings left:",
+          get_guessed_word(secret_word, letters_guessed))
+    return warnings_left
+
+def not_correct_input_losing_guesses(guesses_left, secret_word, letters_guessed):
+    '''
+    System of losing guesses when letter wasn't input and not having warnings left.
+    '''
+    guesses_left -= 1
+    print("Oops! That is not a valid letter. You have no warnings left so you lose one guess:",
+          get_guessed_word(secret_word, letters_guessed))
+    return guesses_left
+
+def guessed_before_losing_warnings(warnings_left, letters_guessed):
+    '''
+    System of losing warnings for entering letter which was guessed before
+    '''
+    warnings_left -= 1
+    print("Oops! You've already guessed that letter. You have", warnings_left, "warnings left:",
+          get_guessed_word(secret_word, letters_guessed))
+    return warnings_left
+
+def guessed_before_losing_guesses(guesses_left, letters_guessed):
+    '''
+    System of losing guesses for entering letter which was guessed before and not having
+    warnings left.
+    '''
+    guesses_left -= 1
+    print("Oops! You've already guessed that letter. You have no warnings left so you lose one guess:",
+          get_guessed_word(secret_word, letters_guessed))
+    return guesses_left
+
+
+def results(guessed_letter, guesses_left, letters_in_secret_word, secret_word, letters_guessed):
+    '''
+    Showing results of current guessing
+    '''
+    if guessed_letter in letters_in_secret_word:
+        # Part of code for correct guess
+        print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+        return guesses_left
+    else:
+        # System of losing guesses for incorrect guessed word
+        if guessed_letter in EN_VOWELS:
+            guesses_left -= 2
+        else:
+            guesses_left -= 1
+        print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+        return guesses_left
+
+
+def end_of_the_game(winning, guesses_left, secret_word):
+    '''
+    Showing information about game results
+    '''
+    if winning:
+        total_score = guesses_left * len(set(secret_word))
+        print("Congratulations, you won! Your total score for this game is:", total_score)
+    else:
+        print("Sorry, you ran out of guesses. The word was", secret_word)
+
+
+def main_match_win_or_lose(guesses_left, letters_guessed, warnings_left, letters_in_secret_word, hints_switch):
```

Comment:
You could pass here only secret_word, not a letters_in_secret_word

Reply:
done